go/types/validating_decoder.go

```diff
@@ -14,9 +14,7 @@ type ValidatingDecoder struct {
 }
 
 func NewValidatingDecoder(cs chunks.ChunkStore) *ValidatingDecoder {
-	return &ValidatingDecoder{
-		newLocalValueStore(cs),
-	}
+	return &ValidatingDecoder{NewValueStore(cs)}
```

Comment:
can we just add this concern into ValidatingChunkStore and give it a flag for whether it or not it should trust that chunks are encoded correctly? For the server, it'd mean that it could also just use a ValidatingChunkStore and also that each chunk would only need to be decoded once.